src/hotspot/cpu/x86/stubGenerator_x86_64_poly_mont.cpp

```diff
@@ -0,0 +1,429 @@
+ *                                                     +--+--+--+--+--+--+--+--+
+ *                                               Acc2+=| 0| 0| 0| 0| 0| 0| 0| 0|
+ *                                                   *h| 0| 0| 0|a5|a4|a3|a2|a1|
+ *          Acc2 += A *h B                             |bi|bi|bi|bi|bi|bi|bi|bi|
+ *                                                     +--+--+--+--+--+--+--+--+
+ *          N = replicate(Acc1[0])                     |n0|n0|n0|n0|n0|n0|n0|n0|
+ *                                                     +--+--+--+--+--+--+--+--+
+ *                                                     +--+--+--+--+--+--+--+--+
+ *                                               Acc1+=| 0| 0| 0|c5|c4|c3|c2|c1|
+ *                                                    *| 0| 0| 0|m5|m4|m3|m2|m1|
+ *          Acc1 += M *  N                             |n0|n0|n0|n0|n0|n0|n0|n0| Note: 52 low bits of Acc1[0] == 0 due to Montgomery!
+ *                                                     +--+--+--+--+--+--+--+--+
+ *                                               Acc2+=| 0| 0| 0|d5|d4|d3|d2|d1|
+ *                                                   *h| 0| 0| 0|m5|m4|m3|m2|m1|
+ *          Acc2 += M *h N                             |n0|n0|n0|n0|n0|n0|n0|n0|
+ *                                                     +--+--+--+--+--+--+--+--+
+ *          if (i == 4) break;
+ *          // Combine high/low partial sums Acc1 + Acc2
+ *                                                     +--+--+--+--+--+--+--+--+
+ *          carry = Acc1[0] >> 52                      | 0| 0| 0| 0| 0| 0| 0|c1|
+ *                                                     +--+--+--+--+--+--+--+--+
+ *          Acc2[0] += carry
+ *                                                     +--+--+--+--+--+--+--+--+
+ *          Acc1 = Acc1 shift one q element>>          | 0| 0| 0| 0|c5|c4|c3|c2|
+ *                                                     +--+--+--+--+--+--+--+--+
+ *          Acc1 = Acc1 + Acc2
+ *      ---- done
+ *   // Last Carry round: Combine high/low partial sums Acc1<high_bits> + Acc1 + Acc2
+ *   carry = Acc1 >> 52
+ *   Acc1 = Acc1 shift one q element >>
+ *   Acc1  = mask52(Acc1)
+ *   Acc2  += carry
+ *   Acc1 = Acc1 + Acc2
+ *   output to rLimbs
+ */
+void StubGenerator::montgomeryMultiply(const Register aLimbs, const Register bLimbs, const Register rLimbs) {
+  Register t0 = r13;
+  Register rscratch = r13;
+
+  // Inputs
+  XMMRegister A = xmm0;
+  XMMRegister B = xmm1;
+  XMMRegister T = xmm2;
+
+  // Intermediates
+  XMMRegister Acc1 = xmm10;
+  XMMRegister Acc2 = xmm11;
+  XMMRegister N = xmm12;
+  XMMRegister select = xmm13;
+  XMMRegister carry = xmm14;
+
+  // Constants
+  XMMRegister modulus = xmm20;
+  XMMRegister shift1L = xmm21;
+  XMMRegister shift1R = xmm22;
+  XMMRegister mask52 = xmm23;
+  XMMRegister broadcast5 = xmm24;
+  KRegister limb0 = k1;
+  KRegister limb5 = k2;
```

Comment:
limb5 and select are not being used anymore.

Reply:
Thanks, fixed (and also broadcast5)